server/test/views/components/PageNotProductionBannerTest.java

```diff
@@ -0,0 +1,150 @@
+package views.components;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.Optional;
+import org.junit.Before;
+import org.junit.Test;
+import play.i18n.Lang;
+import play.i18n.Langs;
+import play.i18n.Messages;
+import play.i18n.MessagesApi;
+import play.mvc.Http;
+import services.DeploymentType;
+import services.MessageKey;
+import services.settings.SettingsManifest;
+
+public class PageNotProductionBannerTest {
+  SettingsManifest settingsManifest;
+  Http.Request request;
+  Messages messages;
+
+  @Before
+  public void setUp() {
+    settingsManifest = mock(SettingsManifest.class);
+    request = mock(Http.Request.class);
+
+    // Setup messages
+    Langs langs = new Langs(new play.api.i18n.DefaultLangs());
+    Map<String, String> messagesMap = new java.util.HashMap<>();
+    messagesMap.put(MessageKey.NOT_FOR_PRODUCTION_BANNER_LINE_1.getKeyName(), "line1");
+    messagesMap.put(MessageKey.NOT_FOR_PRODUCTION_BANNER_LINE_2.getKeyName(), "line2 {0}");
+
+    Map<String, Map<String, String>> langMap =
+        Collections.singletonMap(Lang.defaultLang().code(), messagesMap);
+    MessagesApi messagesApi = play.test.Helpers.stubMessagesApi(langMap, langs);
+    messages = messagesApi.preferred(langs.availables());
+  }
+
+  @Test
+  public void devEnvironment_withShowBannerSettingSettingDisabled_returnsEmpty() {
+    DeploymentType deploymentType = new DeploymentType(/* isDev */ true, /* isStaging */ false);
+    assertEnvironmentWithShowBannerSettingDisabledReturnsEmpty(deploymentType);
+  }
+
+  @Test
+  public void stagingEnvironment_withShowBannerSettingSettingDisabled_returnsEmpty() {
+    DeploymentType deploymentType = new DeploymentType(/* isDev */ false, /* isStaging */ true);
+    assertEnvironmentWithShowBannerSettingDisabledReturnsEmpty(deploymentType);
+  }
+
+  @Test
+  public void productionEnvironment_withShowBannerSettingSettingDisabled_returnsEmpty() {
+    DeploymentType deploymentType = new DeploymentType(/* isDev */ false, /* isStaging */ false);
+    assertEnvironmentWithShowBannerSettingDisabledReturnsEmpty(deploymentType);
+  }
+
+  private void assertEnvironmentWithShowBannerSettingDisabledReturnsEmpty(
```

Comment:
**nitpick:** I usually like when the functions are at the bottom of the file to make it easier to parse where the tests end